### PR TITLE
Update `vhlo.md` with contribution guidelines.

### PR DESCRIPTION
**Pull request by GleasonK**
Add guidelines for implementing compatibility for new StableHLO features in VHLO. Will be useful to link to this doc in future reviews.

This does not impact anything about the compatibility process in `compatibility.md` or the Compatibility RFC, this is intended to aid developers.

### CODE REVIEW

File: docs/vhlo.md [hunk 1]
```diff
@@ -71,3 +71,76 @@ frameworks (producers) only need to target StableHLO ops, and compiler
...
+
+### Add Required VHLO Implementation and Conversions
+
+The exact code needed to integrate a new feature will vary, but for the most
+part the following will need to change:
+
+* For all changes:
+  * Update the version log in [VhloDialect.td](https://github.com/openxla/stablehlo/blob/main/stablehlo/dialect/VhloDialect.td#L29)
+* For new ops:
+  * Add the op in [VhloOps.h](https://github.com/openxla/stablehlo/blob/main/stablehlo/dialect/VhloOps.h)
```
Comment (burmako) [inline docs/vhlo.md:101]: VhloOps.td? (Same below).

File: docs/vhlo.md [hunk 2]
```diff
@@ -71,3 +71,76 @@ frameworks (producers) only need to target StableHLO ops, and compiler
...
+  * Add StableHLO → VHLO conversion in [StablehloLegalizeToVhlo.cpp](https://github.com/openxla/stablehlo/blob/main/stablehlo/transforms/StablehloLegalizeToVhlo.cpp)
+  * Add VHLO → StableHLO conversion in [VhloLegalizeToStablehlo.cpp](https://github.com/openxla/stablehlo/blob/main/stablehlo/transforms/VhloLegalizeToStablehlo.cpp)
+* For new versions of existing ops:
+  * Add the op in [VhloOps.h](https://github.com/openxla/stablehlo/blob/main/stablehlo/dialect/VhloOps.h)
+  * Update StableHLO to VHLO mapping in [MapStablehloToVhlo.h](https://github.com/openxla/stablehlo/blob/main/stablehlo/transforms/MapStablehloToVhlo.h)
+  * Add a conversion between new and old op versions in [VhloToVersion.cpp](https://github.com/openxla/stablehlo/blob/main/stablehlo/transforms/VhloToVersion.cpp)
+* For new types or attributes:
+  * Add the type in [VhloTypes.td](https://github.com/openxla/stablehlo/blob/main/stablehlo/dialect/VhloTypes.td)
+  or [VhloAttrs.td](https://github.com/openxla/stablehlo/blob/main/stablehlo/dialect/VhloAttrs.td)
```
Comment (burmako) [inline docs/vhlo.md:110]: "or the attribute to..."?

File: docs/vhlo.md [hunk 3]
```diff
@@ -71,3 +71,76 @@ frameworks (producers) only need to target StableHLO ops, and compiler
...
+
+A recent example of a compatibility related submission was the addition of two
+FP8 types, as well as their implementation in VHLO in [#1379](https://github.com/openxla/stablehlo/pull/1379).
+
+### Add / Update Unit Tests
+
+The contributor of an incompatible change is responsible for both positive and
+negative unit tests of the feature, as well as compatibility unit tests.
+
+Compatibility unit testing involves updating [stablehlo_legalize_to_vhlo.mlir](FIXME)
```
Comment (burmako) [inline docs/vhlo.md:122]: FIXME?

**Review by burmako — CHANGES_REQUESTED** [review]
A few minor comments

File: docs/vhlo.md (hunk 1 above)
Comment (burmako) [reply docs/vhlo.md:101]: I don't think this has been addressed. Maybe forgot to commit and/or push?

**Review by burmako — APPROVED** [review]
LGTM once the final comment is addressed

File: docs/vhlo.md (hunk 1 above)
Comment (GleasonK) [reply docs/vhlo.md]: Ah. This exact issue occurred twice, I fixed the other one previously. Fixed both now, thanks

**Pull request merged**